### PR TITLE
ref(smartsearch): improve keyboard navigation

### PR DESCRIPTION
**Pull request by JonasBa**
I have been using the smart search more recently and noticed a couple of quirks when using only keyboard navigation to interface with the component. My use case is using the mouse to focus on the input, selecting a filter and value using keyboard navigation and then closing the dropdown and triggering a search. 

https://user-images.githubusercontent.com/9317857/175347638-0949ca12-c986-430c-86f4-2a2d9bc4edec.mp4

From ^, I find it confusing that first escape keydown only clears the current selection from the list, but keeps the dropdown open and the second escape calls blur on the input.

The [aria spec](https://w3c.github.io/aria-practices/#combobox) has the following recommendations:
```
When focus is in the combobox:
...
Escape: Dismisses the popup if it is visible. Optionally, if the popup is hidden before Escape is pressed, clears the combobox.
```
```
When focus is in a listbox popup:
...
Escape: Closes the popup and returns focus to the combobox. Optionally, if the combobox is editable, clears the contents of the combobox.
```

I think what I found counterintuitive was that pressing enter will fire a query **only if you are not currently selecting any items** in the dropdown which I didn't know of because it's not a common pattern. So what I ended up doing was pressing escape twice (at which point I lost focus on search so I could no longer press enter to apply my query and I had to click to focus again)

My change preserves the original functionality, but it also keeps the box more in line with the aria spec by adding the closing of dropdown mechanism and preserving focus, allowing users to hit enter to more explicitly apply the query.

New UX:

https://user-images.githubusercontent.com/9317857/175354241-0340cc9c-3b51-4ada-9062-684e867c54d3.mp4

Note: I havent bothered adding/updating tests yet, because I want to hear opinions first. There is also a bug where hitting enter will briefly display the dropdown which should not happen.

### CODE REVIEW

File: static/app/components/smartSearchBar/index.tsx [hunk 1]
```diff
@@ -72,6 +72,8 @@ const ACTION_OVERFLOW_STEPS = 75;
 
 const makeQueryState = (query: string) => ({
   query,
+  // Anytime the query changes and it is not "" the dropdown should show
+  showDropdown: !!query,
```
Comment (jennmueng) [inline static/app/components/smartSearchBar/index.tsx:76]: We should have the dropdown show on the empty query so we can show the default state. So I guess have it show every time the query is changed instead?

```suggestion
  showDropdown: true,
```

File: static/app/components/smartSearchBar/index.tsx [hunk 2]
```diff
@@ -1450,6 +1465,10 @@ class SmartSearchBar extends Component<Props, State> {
     this.onAutoCompleteFromAst(replaceText, item);
   };
 
+  get showSearchDropdown(): boolean {
+    return this.state.loading || this.state.searchGroups.length > 0;
```
Comment (jennmueng) [inline static/app/components/smartSearchBar/index.tsx:1469]: This isn't used anywhere it looks like. Could there be a possibility of an edge case where there are no search groups but the dropdown is still being displayed?

**Review by jennmueng — COMMENTED** [review]
Wow I really like this. The escape key just hiding the dropdown and not un-focusing on the input so I can hit enter again to submit makes it feel way more natural. One small thing I'd change is the empty state but thanks so much for this!

File: static/app/components/smartSearchBar/index.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/smartSearchBar/index.tsx]: lol, I actually did this change after demoing to @Zylphrex, but havent pushed it. Good eye!

File: static/app/components/smartSearchBar/index.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/smartSearchBar/index.tsx:1469]: Honestly, I dont know enough about it to say. I'll just drop this entirely

**Review by jennmueng — APPROVED** [review]
LGTM but maybe @evanpurkhiser could look at my changes to the tests

**Comment by JonasBa** [comment]
Thanks for fixing the tests @jennmueng 🥇

**Review by evanpurkhiser — APPROVED** [review]
thanks @JonasBa !

**Comment by JonasBa** [comment]
Thanks for the extra pair of eyes @jennmueng @evanpurkhiser. I'll merge the change

**Pull request merged**